File: BOJ/boj18258.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long

int main() {
    cin.tie(nullptr);
    ios::sync_with_stdio(false);
    int t;
    cin >> t;

    queue<int> q;

    string command;
    for(int i = 0; i < t; i++) {
        cin >> command;
        if(command == "push") {
            int num;
            cin >> num;
            q.push(num);
        } else if(command == "pop") {
            if(q.empty()) { cout << "-1\n"; }
            else {
                cout << q.front() << "\n";
                q.pop();
            }
        } else if(command == "size"){
            cout << q.size() << "\n";
        } else if(command == "empty") {
            if(q.empty()) {
                cout << "1\n";
            } else {
                cout << "0\n";
            }
        } else if(command == "front") {
            if(q.empty()) {cout << "-1\n";}
            else {
                cout << q.front() << "\n";
            }
        } else if(command == "back") {
            if(q.empty()) {cout << "-1\n";}
            else {
                cout << q.back() << "\n";
            }
        }
    }

    return 0;
}

File: BOJ/boj18258.java
import java.io.*;
import java.util.*;

/*
 * 18258 큐2
 */
public class boj18258 {
    static StringBuffer sb = new StringBuffer();
    static int n, end = -1;
    static Queue<Integer> queue = new LinkedList<>();

    public static void main(String[] args) throws Exception {
        input();
        //TO-DO
        System.out.println(sb);
    }

    public static void input() throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        Scanner sc = new Scanner(System.in);
        //TO-DO
        // StringTokenizer st = new StringTokenizer(br.readLine());
        n = Integer.parseInt(br.readLine());
        for (int i = 0; i < n; i++) {
            String[] input = br.readLine().split(" ");
            solve(input);
        }
        //END
        sc.close();
		br.close();
    }

    public static void solve(String[] input) {
        if (input[0].equals("push")) {
            int num = Integer.parseInt(input[1]);
            queue.add(num);
            end = num;
        } else if (input[0].equals("pop")) {
            if (queue.isEmpty()) {
                sb.append("-1\n");
            } else {
                sb.append(queue.poll()+"\n");
            }
        } else if (input[0].equals("size")) {
            sb.append(queue.size()+"\n");
        } else if (input[0].equals("empty")) {
            if (queue.isEmpty()) {
                sb.append("1\n");
            } else {
                sb.append("0\n");
            }
        } else if (input[0].equals("front")) {
            if (queue.isEmpty()) {
                sb.append("-1\n");
            } else {
                sb.append(queue.peek()+"\n");
            }
        } else { // back
            if (queue.isEmpty()) {
                sb.append("-1\n");
            } else {
                sb.append(end+"\n");
            }
        }
    }
}
